RIOT-OS/RIOT
# cpu/nrf52: add support for the whole family, fix nrf52811

### Contribution description

The addition of nrf52811 was regrettably untested and it turns out it didn't even compile.
To generalize the fix and avoid the trap of adding CPU-specific `ifdefs`, I added support for the remaining family members (nrf52805/10/20/33) too.

I also noticed the interrupt vector tables for the existing nrf52832 and nrf52840 were wrong.

### Testing procedure

Existing nrf52 boards should still work. Especially SPI & I2C (also both at the same time),

I added a board based on the [E104-BT5011A](http://www.ebyte.com/en/product-view-news.aspx?id=1043) (nrf52810) module.
Such boards can be [bought](https://de.aliexpress.com/item/4000867497654.html) for cheap on AliExpress and similar sites. 

I can add dummy boards for the new family members - according to the documentation they can be simulated on `nrf52832dk`/`nrf52840dk`.

For now, let's see if Murdock is happy with the changes for the existing boards. 

@Citrullin, @vvzvlad do you have a nrf52811 board?

### Issues/PRs references

follow-up to #14008
